File: VB/VB/WinSolution.Module/DomainObject1.vb
Imports Microsoft.VisualBasic
Imports DevExpress.Xpo
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports System

Namespace WinSolution.Module
	<DefaultClassOptions> _
	Public Class DomainObject1
		Inherits BaseObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Private _CreatedOn As DateTime
		Public Property CreatedOn() As DateTime
			Get
				Return _CreatedOn
			End Get
			Set(ByVal value As DateTime)
				SetPropertyValue("CreatedOn", _CreatedOn, value)
			End Set
		End Property
		Private _Name As String
		Public Property Name() As String
			Get
				Return _Name
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Name", _Name, value)
			End Set
		End Property
	End Class
End Namespace

File: CS/E1098/WinSolution.Module/DomainObject1.cs
using DevExpress.Xpo;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using System;

namespace WinSolution.Module {
    [DefaultClassOptions]
    public class DomainObject1 : BaseObject {
        public DomainObject1(Session session) : base(session) { }
        private DateTime _CreatedOn;
        public DateTime CreatedOn {
            get { return _CreatedOn; }
            set { SetPropertyValue("CreatedOn", ref _CreatedOn, value); }
        }
        private string _Name;
        public string Name {
            get { return _Name; }
            set { SetPropertyValue("Name", ref _Name, value); }
        }
    }
}
